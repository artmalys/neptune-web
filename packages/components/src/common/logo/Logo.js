import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';
import { Theme } from '..';
import './Logo.css';

const Logo = ({ className, theme, type }) => {
  let baseClass;
  if (type === Logo.Type.FLAG) {
    baseClass = ['flag', 'flag-info'];
  } else {
    baseClass = ['logo', 'logo-3'];
    if (theme === Logo.Theme.DARK) {
      baseClass.push('logo-inverse');
    } else {
      baseClass.push('logo-primary');
    }
  }
  return <div className={classNames('np-logo', baseClass, className)} />;
};

Logo.Theme = Theme;

Logo.Type = { FULL: 'FULL', FLAG: 'FLAG' };

Logo.propTypes = {
  className: Types.string,
  theme: Types.oneOf([Logo.Theme.LIGHT, Logo.Theme.DARK]),
  type: Types.oneOf([Logo.Type.FULL, Logo.Type.FLAG]),
};

Logo.defaultProps = {
  className: undefined,
  theme: Theme.LIGHT,
  type: Logo.Type.FULL,
};

export default Logo;
