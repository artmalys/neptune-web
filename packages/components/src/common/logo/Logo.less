.np-logo {
  &.flag {
    width: var(--size-24);
    height: var(--size-32);
  }
}

.np-flow-navigation {
  &--large {
    .logo {
      margin-top: -10px;
    }
  }
}
