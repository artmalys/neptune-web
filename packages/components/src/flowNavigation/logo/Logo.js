// To be extracted and refactored as part of https://transferwise.atlassian.net/browse/DS-274
import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';
import { Theme } from '../../common';

const Logo = ({ theme, onGoBack }) => (
  <>
    <div className={classNames('logo', 'hidden-xs')}>
      <img
        src={
          theme === Theme.LIGHT
            ? 'https://transferwise.com/public-resources/assets/logos/transferwise/logo.svg'
            : 'https://transferwise.com/public-resources/assets/logos/transferwise/logo_inverse.svg'
        }
        className="logo__image"
        alt="TransferWise"
      />
    </div>
    <div
      className={classNames('flag', 'visible-xs', {
        'flag--hidden': onGoBack,
      })}
    >
      <img
        src="https://transferwise.com/public-resources/assets/logos/transferwise/flag.svg"
        className="logo__image"
        alt="TransferWise"
      />
    </div>
  </>
);

Logo.defaultProps = {
  theme: Theme.LIGHT,
  onGoBack: null,
};

Logo.propTypes = {
  theme: Types.oneOf([Theme.LIGHT, Theme.DARK]),
  onGoBack: Types.func,
};

export default Logo;
