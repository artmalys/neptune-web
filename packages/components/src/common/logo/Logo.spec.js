import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import Logo from '.';

const props = {
  className: 'className',
  theme: Logo.Theme.LIGHT,
  type: Logo.Type.FULL,
};
describe('Logo', () => {
  it(`renders as expected for theme ${Logo.Theme.LIGHT} and type ${Logo.Type.FULL}`, () => {
    expect(render(<Logo {...props} />).container).toMatchSnapshot();
  });
  it(`renders as expected for theme ${Logo.Theme.LIGHT} and type ${Logo.Type.FLAG}`, () => {
    expect(render(<Logo {...props} type={Logo.Type.FLAG} />).container).toMatchSnapshot();
  });
  it(`renders as expected for theme ${Logo.Theme.DARK} and type ${Logo.Type.FULL}`, () => {
    expect(render(<Logo {...props} theme={Logo.Theme.DARK} />).container).toMatchSnapshot();
  });
});
