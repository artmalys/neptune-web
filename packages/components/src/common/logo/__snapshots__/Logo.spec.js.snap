// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders as expected for theme dark and type FULL 1`] = `
<div>
  <div
    class="np-logo logo logo-3 logo-inverse className"
  />
</div>
`;

exports[`Logo renders as expected for theme light and type FLAG 1`] = `
<div>
  <div
    class="np-logo flag flag-info className"
  />
</div>
`;

exports[`Logo renders as expected for theme light and type FULL 1`] = `
<div>
  <div
    class="np-logo logo logo-3 logo-primary className"
  />
</div>
`;
